docs/service-and-repository-structure/service-and-repository-structure.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Service and repository structure
+---
+
+# 1. Summary
+
+NineChronicles(a.k.a. 9c) is role-playing game on blockchain. To achieve the game on blockchain, 9c runs with multiple
+libraries in certain structure.  
+In this document, you will understand how 9c is structured and interact each other. After that, you would know what
+repository you should touch for your mod/utils.
+
+# 2. Service Structure
+
+![NineChronicles Service Structure](images/0001_service_structure.png)
+
+NineChronicles has 3 parts in major. `Launcher`, `Headless Server`, `Unity Client` are those.  
+In this chapter, We'll see each part in brief.
+
+## 2.1. Launcher
+
+`Launcher` is the one you execute to play game. This checks NineChronicles game version and updates `Launcher`
+and [`Unity Client`](#2.3.-Unity-Client) to the latest version.  
+After login in `Launcher`, the `Launcher` runs Unity client to play game with your logged-in account.  
+`Launcher` also has embedded [`Headless Server`](#2.2.-Headless-Server) to attach to remote RPC node and use GraphQL
+query to get data from server.
```

Comment:
```suggestion
Launcher also has embedded [`Headless Server`](#2.2.-Headless-Server) to read and sign blockchain-related data from the Nine Chronicles network.
```

The embedded headless itself doesn't connect or communicate with an RPC node, but it helps communicate by providing a signing interface for the launcher and sometimes works as a standalone RPC server (which we typically call as a local mode).